website/docs/r/device_security_group.html.markdown

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Security Center"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_device_security_group"
+description: |-
+  Manages a Device Security Group.
+---
+
+# azurerm_device_security_group
+
+Manages a Device Security Group.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_iothub" "example" {
+  name                = "example-IoTHub"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  sku {
+    name     = "S1"
+    capacity = "1"
+  }
+}
+
+resource "azurerm_iot_security_solution" "example" {
+  name                = "example-Iot-Security-Solution"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  display_name        = "Iot Security Solution"
+  iothub_ids          = [azurerm_iothub.example.id]
+}
+
+resource "azurerm_device_security_group" "example" {
+  name               = "example-device-security-group"
+  target_resource_id = azurerm_iothub.example.id
+
+  allow_list_rule {
+    type   = "LocalUserNotAllowed"
+    values = ["user0"]
+  }
+
+  time_window_rule {
+    type             = "ActiveConnectionsNotInAllowedRange"
+    min_threshold    = 0
+    max_threshold    = 30
+    time_window_size = "PT5M"
+  }
+
+  depends_on = [azurerm_iot_security_solution.example]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Device Security Group. Changing this forces a new resource to be created.
+
+* `target_resource_id` - (Required) The ID of the Azure Resource which to create the Device Security Group. Changing this forces a new resource to be created.
+
+* `allow_list_rule` - (Optional) One or more `allow_list_rule` blocks as defined below.
+
+* `time_window_rule` - (Optional) One or more `time_window_rule` blocks as defined below.
+
+---
+
+An `allow_list_rule` block supports the following:
+
+* `type` - (Required) The type of supported rule type. Possible Values are `ConnectionToIpNotAllowed`, `LocalUserNotAllowed` and `LocalUserNotAllowed`.
+
+* `values` - (Required) The values to allow.
```

Comment:
looking at this now would it make sense to add different allow rule type blocks? block_ip_rule? allow_local_user_rule? 

also duplicate value above

Reply:
have made each type a different schema